src/runtime/test/ranger_resource_policy_manager_test.cpp

```diff
@@ -256,5 +257,96 @@ TEST(ranger_resource_policy_manager_test, get_table_name_from_app_name_test)
+        fake_policy_5.allow_policies = {{access_type::kControl, {"user9", "user10"}}};
+        fake_policy_5.allow_policies_exclude = {{access_type::kControl, {"user10"}}};
+        ranger_resource_policy fake_ranger_resource_policy_5;
+        fake_ranger_resource_policy_5.database_names = {"database4"};
+        fake_ranger_resource_policy_5.policies = fake_policy_5;
+        _global_policies_cache = {fake_ranger_resource_policy_4, fake_ranger_resource_policy_5};
+    }
+};
+
+TEST_F(ranger_resource_policy_manager_function_test, allowed)
+{
+    struct test_case
+    {
+        std::string rpc_code;
+        std::string user_name;
+        std::string database_name;
+        bool expected_result;
+    } tests[] = {{"TASK_CODE_INVALID", "user1", "database1", false},
+                 {"RPC_CM_CREATE_APP", "user1", "database1", false},
+                 {"RPC_CM_CREATE_APP", "user2", "database1", false},
+                 {"RPC_CM_LIST_APPS", "user1", "database1", true},
+                 {"RPC_CM_LIST_APPS", "user2", "database1", true},
+                 {"RPC_CM_GET_MAX_REPLICA_COUNT", "user1", "database1", true},
+                 {"RPC_CM_GET_MAX_REPLICA_COUNT", "user2", "database1", false},
+                 {"TASK_CODE_INVALID", "user3", "database2", false},
+                 {"RPC_CM_CREATE_APP", "user3", "database2", true},
+                 {"RPC_CM_CREATE_APP", "user4", "database2", true},
+                 {"RPC_CM_START_BACKUP_APP", "user3", "database2", true},
+                 {"RPC_CM_START_BACKUP_APP", "user4", "database2", false},
+                 {"TASK_CODE_INVALID", "user5", "", false},
+                 {"RPC_CM_CREATE_APP", "user5", "", true},
+                 {"RPC_CM_CREATE_APP", "user5", "database2", false},
+                 {"RPC_CM_CREATE_APP", "user6", "", false},
+                 {"RPC_CM_CREATE_APP", "user6", "database2", false},
+                 {"TASK_CODE_INVALID", "user7", "database3", false},
+                 {"RPC_CM_LIST_NODES", "user7", "database3", true},
+                 {"RPC_CM_LIST_NODES", "user8", "database3", false},
+                 {"RPC_CM_LIST_APPS", "user7", "database3", true},
+                 {"RPC_CM_LIST_APPS", "user8", "database3", false},
+                 {"TASK_CODE_INVALID", "user9", "database4", false},
+                 {"RPC_CM_LIST_NODES", "user9", "database4", false},
+                 {"RPC_CM_LIST_NODES", "user10", "database4", false},
+                 {"RPC_CM_LIST_APPS", "user9", "database4", false},
+                 {"RPC_CM_LIST_APPS", "user10", "database4", false},
+                 {"RPC_CM_CONTROL_META", "user9", "database4", true},
+                 {"RPC_CM_CONTROL_META", "user10", "database4", false}};
+    for (const auto &test : tests) {
+        auto code = task_code::try_get(test.rpc_code, TASK_CODE_INVALID);
+        auto actual_result = allowed(code, test.user_name, test.database_name);
+        EXPECT_EQ(test.expected_result, actual_result);
```

Comment:
How about output parameters like the following, so if any case failed, you can find out which case failed immediately.
```
EXPECT_EQ(test.expected_result, actual_result) << fmt::format("rpc_code: {}, user_name: {}, database_name: {}", rpc_code, user_name, database_name);
```